internals/plan/plan.go

```diff
@@ -1021,82 +1022,111 @@ func validServiceAction(action ServiceAction, additionalValid ...ServiceAction)
 
 var fnameExp = regexp.MustCompile("^([0-9]{3})-([a-z](?:-?[a-z0-9]){2,}).yaml$")
 
-func ReadLayersDir(dirname string) ([]*Layer, error) {
+type LayerReader struct {
+	labels map[string]string
+	offset int
+}
+
+func (lr *LayerReader) ReadDir(dirname string) ([]*Layer, error) {
+	if lr.labels == nil {
+		lr.labels = make(map[string]string)
+	}
+
 	finfos, err := ioutil.ReadDir(dirname)
 	if err != nil {
 		// Errors from package os generally include the path.
 		return nil, fmt.Errorf("cannot read layers directory: %v", err)
 	}
 
 	orders := make(map[int]string)
-	labels := make(map[string]int)
+	labels := make(map[string]string)
+	// Copy already loaded layer labels with full path to the layer file
+	// as they have come from a different import path.
+	for k, v := range lr.labels {
+		labels[k] = v
```

Comment:
Why can't we operate on `lr.labels` directly below, instead of creating a copy?

Reply:
As discussed, this was to simplify duplicate label errors. Since we probably want the full path anyway, I've simplified this.